app/controllers/active_storage/blobs/proxy_controller.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+# Proxy files through application with authentication.
+# This avoids making the S3 bucket where files are stored public
+
+# Overwriting
+# https://github.com/rails/rails/blob/6-1-stable/activestorage/app/controllers/active_storage/blobs/proxy_controller.rb
+class ActiveStorage::Blobs::ProxyController < ActiveStorage::BaseController
+  include ActiveStorage::SetBlob
+  include ActiveStorage::SetHeaders
+  include ActionController::Live
+
+  before_action :authenticate_user!, :authorize_user!
+
+  def show
+    http_cache_forever public: true do
+      set_content_headers_from @blob
+      stream @blob
+    end
+  end
```

Comment:
Need to do something to queue files for deletion after they are fully downloaded

Reply:
https://github.com/SaraAlert/SaraAlert/pull/550/commits/cfd5cbe70220f1b9c9d7aee7baaf4feb8e6cf5ba

Reply:
verified that file is marked for deletion upon download

Reply:
verified that file destroy job is queued and run at the appropriate time and that the file is removed from S3